content/projects/gsoc/2020/project-ideas/jenkins-distribution-customize-service.adoc

```diff
@@ -0,0 +1,41 @@
+---
+layout: gsocprojectidea
+title: "Jenkins distribution customize service"
+goal: "Provide an out of the box solution for Jenkins users"
+category: Service
+year: 2020
+status: draft
+skills:
+- Java
+- Web Development
+mentors:
+- "linuxsuren"
+---
+
+For many users, they download Jenkins first, then select some plugins and config them. 
+It might takes a lot of time, like hours. But if we can get a perfect Jenkins distribution which contains all we need, 
+it can save that time for us.
+
+The service could like this, a website offered as https://customize.jenkins.io. Plus, it should support self-host. 
+People can select the following configuration items:
+
+* Jenkins core version
+* plugins
+* common configuration, user/password, update-center site e.g.
+* plugin based configuration, Kubernetes, Sonarqube plugin config e.g.
+* multi-form package, jenkins.war or docker image
+* other things
+
+Two projects are expected backend and frontend. Thanks to link:https://github.com/jenkinsci/custom-war-packager[Custom Jenkins WAR packager for Jenkins], 
+it already did a lot of works. It can be the main library for the backend project. 
+
+The backend should provide the modern Restful API. We can develop different client base on that, UI or CLI. A readable API document is
+necessary. Like many projects, the Swagger UI is a good example.
+We can start the backend project with SpringBoot, it might save a lot of time.
```

Comment:
One of the things to consider is a front end UI, could be react or even angular.

Reply:
I don't have enough experience with the front-end. Any ideas are welcome.

Reply:
Blue Ocean was using React, so as of last year, that would have seemed like the right choice. I'm not sure what's up with Blue Ocean these days, so maybe there'll be a new UI framework?

Ignoring that choice, React seems to be more popular, and also newer FWIW. We switched all of our front end from Angular to React a couple years back.

Reply:
Yeah React/Angular seem to be the most popular choices for a frontend. Some pros in favour of react (Personal Experience here :) ) 

1. Single Way DataFlow ( Events Percolate upwards. Something which we need) .
2. Testablility
3. Huge Community Support.
4. Performance